ssl/record/rec_layer_s3.c

```diff
@@ -268,11 +268,13 @@ int ssl3_read_n(SSL *s, size_t n, size_t max, int extend, int clearold,
         return -1;
     }
 
-    /* We always act like read_ahead is set for DTLS */
-    if (!s->rlayer.read_ahead && !SSL_IS_DTLS(s))
+    /* Ktls always reads full records.
+     * Also, we always act like read_ahead is set for DTLS */
+    if (!BIO_get_ktls_recv(s->rbio) && !s->rlayer.read_ahead &&
+        !SSL_IS_DTLS(s)) {
```

Comment:
Please break lines before the `&&` operator

Reply:
Done